M-37 Weather CSS Refactor

chrisBRN opened the pull request.

MichaelWalker commented on src/components/weather-page/SingleDayDetail.tsx:
```diff
@@ -0,0 +1,48 @@
+export default function SingleDayDetail(props : SingleDayData) {  
+    
+    let min = props.minTemp ? Math.round(props.minTemp) + " °C" : 0 + " °C";   
+    let max = props.maxTemp ? Math.round(props.maxTemp) + " °C": 0 + " °C"; 
```
little bit of duplicated code here - also not sure that `0` is a good option for no data, as it looks the same as actually having a temp of 0.

How about

```ts
function formatTemperature(temperature?: number): string {
    if (!number) {
        return "N/A";
    }
    return `${Math.round(temperature)} °C`;
}

const min = formatTemperature(props.minTemp);
const max = formatTemperature(props.maxTemp);
```

MichaelWalker replied:
We can move `formatTemperature` into its own helper function too, and add some unit tests for it.

chrisBRN replied:
Thanks Mike, does the !number know the variable name by magic (genuine question) or is that a typo?

chrisBRN replied:
No worries either way, I wasn't sure on the "$" syntax either so went with 

```

export default function formatTemperature (temperature?: number) : string{
    return temperature ? Math.round(temperature) + ' °C' : "N/A";
}

````

MichaelWalker replied:
`number` was just a typo.

the `$` syntax is [JS string literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

chrisBRN commented:
Hi Mike

Not sure why the Weather.scss is causing the tests to fail, we updated the capitalisation of all out files earlier but this one seems to have been missed on the commit/push since it reverted to lower case at home. I've updated the file here but its still complaining.

Chris

MichaelWalker commented:
At a guess for the failing tests...

Windows doesn't care about capitalisation on file names.
Linux does.

Windows will therefore ignore a commit that just changes the capitalisation of a file.

If you need to change it, I've not found a better way than
- Changing the filename to something different
- Committing that
- Changing it back to what you want (with the correct capitalisation)
- Committing that.

MichaelWalker commented on src/helpers/FormatTemperature.ts:
```diff
@@ -0,0 +1,4 @@
+export default function formatTemperature (temperature?: number) : string{
+    return temperature ? Math.round(temperature) + ' °C' : "N/A";
+}
+
```
Could we add a couple of tests for this?

chrisBRN replied:
Shouldn't be an issue, probably a new ticket though? since this is all under CSS refactor...?

MichaelWalker replied:
This is the PR where we've created this function, so we should really test it at the same time.

MichaelWalker commented on src/helpers/FormatTemperature.test.tsx:
```diff
@@ -0,0 +1,12 @@
+    
+    expect(formatTemperature("dave")).toBe("N/A");
+
+  });
```
Unit tests would ideally test all of the cases that we care about, so in this case it would be something like;

```ts
describe('Format Temperature', () => {
    it('should round a number and convert to string with unit', () => {
        // test code goes here
    }); 

    it('should return N/A if the input is null', () => {
        // test code goes here
    });
});
```
